EmberAPI/clsAPIYouTube.vb

```diff
@@ -140,14 +141,26 @@ Namespace YouTube
                     Dim regSearchResults As MatchCollection = Regex.Matches(regSearchResultsArea.Item(0).Value, strVideoIdPattern, RegexOptions.Singleline, TimeSpan.FromSeconds(1))
                     Dim lstVideoIds As New List(Of String)
                     For ctr = 0 To regSearchResults.Count - 1
-                        lstVideoIds.Add(regSearchResults.Item(ctr).Groups(1).Value)
-                    Next
-                    'clean the list from duplicate entries
-                    lstVideoIds = lstVideoIds.Distinct.ToList
-                    For Each tId In lstVideoIds
-                        Dim nVideoDetails = GetVideoDetails(tId)
-                        If nVideoDetails IsNot Nothing Then nVideoList.Add(nVideoDetails)
+                        Dim tId As String
+                        tId = regSearchResults.Item(ctr).Groups(1).Value
+                        'prevent duplicate entries
+                        If (Not lstVideoIds.Contains(tId)) Then lstVideoIds.Add(tId)
                     Next
+                    Dim nVideoDict As New SortedDictionary(Of Long, MediaContainers.MediaFile)
```

Comment:
need SortedDict to keep existing order - youtube sorts by relevance which is good